### PR TITLE
icu: Also provide ICU name

### PR DESCRIPTION
**Pull request by xclaesse**
That's the CMake name that harfbuzz lookup when using MSVC.

### CODE REVIEW

**Comment by xclaesse** [comment]
See how harfbuzz lookup for ICU: https://github.com/harfbuzz/harfbuzz/blob/main/meson.build#L96

**Comment by xclaesse** [comment]
CI fails because of https://github.com/mesonbuild/meson/pull/10789

**Comment by eli-schwartz** [comment]
I don't like this change, nor the other one. The correct solution is for harfbuzz to manually specify an old style fallback: kwarg.

File: subprojects/icu.wrap [hunk 1]
```diff
@@ -6,6 +6,7 @@ source_hash = 67a7e6e51f61faf1306b6935333e13b2c48abd8da6d2f46ce6adca24b1e21ebf
 patch_directory = icu
 
 [provide]
+ICU = icuuc_dep
```
Comment (eli-schwartz) [inline subprojects/icu.wrap:9]: This is outright broken, the cmake.org published FindICU.cmake module (not officially part of the icu project) exports ICU::uc, as well as ICU::io, ICU::i18n, so in order to use cmake lookups you need to pass a `modules:` kwarg, and there's a fairly good chance the user didn't want icu-uc specifically.

**Review by eli-schwartz — CHANGES_REQUESTED** [review]

File: subprojects/icu.wrap (hunk 1 above)
Comment (xclaesse) [reply subprojects/icu.wrap:9]: Hmm, they indeed pass a component: https://github.com/harfbuzz/harfbuzz/blob/main/meson.build#L96. Our `[provide]` mechanism cannot deal with components, so I guess there is no solution here :/

**Pull request closed without merging**